sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuArrowEvalPythonExec.scala

```diff
@@ -0,0 +1,489 @@
+        assert(children.forall(_.find(_.isInstanceOf[GpuPythonUDF]).isEmpty))
+        (ChainedPythonFunctions(Seq(udf.func)), udf.children)
+    }
+  }
+
+  private val batchSize = conf.arrowMaxRecordsPerBatch
+  private val sessionLocalTimeZone = conf.sessionLocalTimeZone
+  private val pythonRunnerConf = ArrowUtils.getPythonRunnerConfMap(conf)
+
+
+  override protected def doExecute(): RDD[InternalRow] =
+    throw new IllegalStateException(s"Row-based execution should not occur for $this")
+
+  override lazy val metrics: Map[String, SQLMetric] = Map(
+    NUM_OUTPUT_ROWS -> SQLMetrics.createMetric(sparkContext, DESCRIPTION_NUM_OUTPUT_ROWS),
+    NUM_OUTPUT_BATCHES -> SQLMetrics.createMetric(sparkContext, DESCRIPTION_NUM_OUTPUT_BATCHES),
+    NUM_INPUT_ROWS -> SQLMetrics.createMetric(sparkContext, DESCRIPTION_NUM_INPUT_ROWS),
+    NUM_INPUT_BATCHES -> SQLMetrics.createMetric(sparkContext, DESCRIPTION_NUM_INPUT_BATCHES)
+  )
+
+  override protected def doExecuteColumnar(): RDD[ColumnarBatch] = {
+    val numOutputRows = longMetric(NUM_OUTPUT_ROWS)
+    val numOutputBatches = longMetric(NUM_OUTPUT_BATCHES)
+    val numInputRows = longMetric(NUM_INPUT_ROWS)
+    val numInputBatches = longMetric(NUM_INPUT_BATCHES)
+
+    val inputRDD = child.executeColumnar()
+    inputRDD.mapPartitions { iter =>
+      val queue: BatchQueue = new BatchQueue()
+      val context = TaskContext.get()
+      context.addTaskCompletionListener[Unit](_ => queue.close())
+
+      val (pyFuncs, inputs) = udfs.map(collectFunctions).unzip
+
+      // Not sure why we are doing this in every task.  It is not going to change, but it might
+      // just be less that we have to ship.
+
+      // flatten all the arguments
+      val allInputs = new ArrayBuffer[Expression]
+      // TODO eventually we should just do type checking on these, but that can get a little complex
+      // with how things are setup for replacement...
+      // perhaps it needs to be with the special, it is an gpu compatible expression, but not a
+      // gpu expression...
+      val dataTypes = new ArrayBuffer[DataType]
+      val argOffsets = inputs.map { input =>
+        input.map { e =>
+          if (allInputs.exists(_.semanticEquals(e))) {
+            allInputs.indexWhere(_.semanticEquals(e))
+          } else {
+            allInputs += e
+            dataTypes += e.dataType
+            allInputs.length - 1
+          }
+        }.toArray
+      }.toArray
+
+      val schema = StructType(dataTypes.zipWithIndex.map { case (dt, i) =>
+        StructField(s"_$i", dt)
+      })
+
+      val boundReferences = GpuBindReferences.bindReferences(allInputs, child.output)
+      val batchedIterator = new RebatchingIterator(iter, batchSize, numInputRows, numInputBatches)
+      val projectedIterator = batchedIterator.map { batch =>
+        queue.add(batch)
+        GpuProjectExec.project(batch, boundReferences)
+      }
+
+      GpuPythonHelper.injectGpuInfo(pyFuncs)
```

Comment:
Do we want this on all of the time? I can see situations where someone wants to speed up the transition to/from python, but not use a GPU as a part of the processing.  It would be good to have a global config of some kind that disables this part of it along with the `PythonWorkerSemaphore`.  In the future it would be even better if we could somehow let users tag a UDF if they want a specific resource or not in it.  But that would have to be follow on work that we would push into spark eventually.

Reply:
Added a new config to enable/disable this feature and applied to both of them.